### DISCUSSION TITLE
How to make `rich.Table.grid` size rigid?

### DISCUSSION THREAD

**Discussion by yashrathi-git** [Textualize/rich#1372] [Q&A] [answered]
Hi 
first of all thank you for creating `rich`. I really enjoy using it.

I am trying to create a time-tracking python CLI app. I got inspired by [timewarrior](https://timewarrior.net/) to do something like this:
![](https://timewarrior.net/images/swatch.png)

I tried to implement it with `rich.Table.grid`, it seems perfect solution. Is there any way to only let columns expand up to a certain limit or make them rigid (content inside get appropriately truncated)?
![image](https://user-images.githubusercontent.com/57002207/127360669-bf5e9601-9b33-45fe-8d32-9330a507c601.png)

Also for multiple tasks in same hour I think `Columns` should be used. If there's any better way please guide me.

Thank you

**Comment by willmcgugan** [comment, marked as answer]
Try this...

```python
from rich.table import Table
from rich import print
t = Table.grid(expand=True)
t.add_column(ratio=1, no_wrap=True)
t.add_column(ratio=1, no_wrap=True)
t.add_row("foo " * 20, "bar " * 20)
print(t)
```

This should create equal size columns that don't wrap.
